src/Microsoft.AspNetCore.Mvc.Abstractions/Filters/FilterContext.cs

```diff
@@ -33,5 +33,67 @@ public abstract class FilterContext : ActionContext
         /// Gets all applicable <see cref="IFilterMetadata"/> implementations.
         /// </summary>
         public virtual IList<IFilterMetadata> Filters { get; }
+
+        /// <summary>
+        /// Returns a value indicating whether the provided <see cref="IFilterMetadata"/> is the most effective
+        /// policy (most specific) applied to the action associated with the <see cref="FilterContext"/>.
+        /// </summary>
+        /// <typeparam name="TMetadata">The type of the filter policy.</typeparam>
+        /// <param name="policy">The filter policy instance.</param>
+        /// <returns>
+        /// <c>true</c> if the provided <see cref="IFilterMetadata"/> is the most effective policy, otherwise <c>false</c>.
+        /// </returns>
+        /// <remarks>
+        /// <para>
+        /// The <see cref="IsEffectivePolicy{TMetadata}(TMetadata)"/> method is used to implement a common convention
+        /// for filters where when that define an overriding behavior. When multiple filters may apply to the same 
+        /// cross-cutting concern, define a common interface for the filters (<typeparamref name="TMetadata"/>) and 
+        /// implement the filters such that all of the implementations call this method to determine if they should
+        /// take action.
+        /// </para>
+        /// <para>
+        /// For instance, a global filter might be overridden by placing a filter attribute on an action method.
+        /// The policy applied directly to the action method could be considered more specific.
+        /// </para>
+        /// <para>
+        /// This mechanism for overriding relies and the rules of order and scope that the filter system
+        /// provides to control ordering of filters. It is up to the implementor of filters to implement this 
+        /// protocol cooperatively. The filter system has no innate notion of overrides, this is a recommended
+        /// convention.
+        /// </para>
+        /// </remarks>
+        public bool IsEffectivePolicy<TMetadata>(TMetadata policy) where TMetadata : IFilterMetadata
```

Comment:
A user would still need to check for this explicitly though. Could we introduce a new marker interface for this scenario (something like IClosestFilterWins or something)? So a user writing a filter can just implement this marker interface and we take care of the logic internally and pass around the 'final' list of filters through a new property like 'EffectiveFilters' or something on the FilterContext?

Reply:
No, I want to keep this as a convention. We decided early on not to build filter overrides into the system

Reply:
Sure 👍 . Just wondering if finding the effective filter should be driven by 'policy' alone. For example, could we provide an additional overload which checks the current type of the filter alone. This way a user doesn't need to define a new type for 'policy'.